src/app/api/semesters/route.ts

```diff
@@ -62,6 +63,18 @@ export async function POST(req: NextRequest) {
       });
     }
 
+    const findName = await db
+      .select()
+      .from(semesters)
+      .where(eq(semesters.name, newSemester.name))
+      .limit(1);
+
+    if (findName.length != 0) {
+      return new Response("This name already exists, please pick another", {
+        status: 400,
+      });
+    }
+
```

Comment:
rename this to something better like `existingSemester` please. This gives more context to what the variable actually is and allows follow up code to be more readable as well. e.g.

```
if (!existingSemester) {
    ...
}
```

to get a single semester instead of a array you could destructure it like so

```
const [existingSemester] = await db.select().....
```

or use the `findFirst` function provided by the query api

```
const existingSemester = await db.query.semesters.findFirst({
    where: ...
})
```